### PR TITLE
build!: Update to Node v22

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Updates Dockerfile and Actions to use Node 22

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
```
# Error: Cannot find module '/home/runner/work/renovate/renovate/tools/docs/test'
#     at Function._resolveFilename (node:internal/modules/cjs/loader:1249:15)
#     at Function._load (node:internal/modules/cjs/loader:1075:27)
#     at TracingChannel.traceSync (node:diagnostics_channel:315:14)
#     at wrapModuleLoad (node:internal/modules/cjs/loader:218:24)
#     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:170:5)
#     at node:internal/main/run_main_module:[36](https://github.com/renovatebot/renovate/actions/runs/11558770142/job/32171997068?pr=32182#step:7:37):49 {
#   code: 'MODULE_NOT_FOUND',
#   requireStack: []
```

@viceice do you know what we need to do about the .mjs file in Node v22?

**Comment by viceice** [comment]
will check later

**Review by viceice — COMMENTED** [review]
wrong changes in this pr

File: package.json [hunk 1]
```diff
@@ -135,11 +135,11 @@
   },
   "homepage": "https://renovatebot.com",
   "engines": {
-    "node": "^20.15.1",
+    "node": "^20.15.1 || ^22.11.0",
     "pnpm": "^9.0.0"
   },
-  "volta": {
-    "node": "20.18.0",
+  "volta":{
```
Comment (rarkins) [inline package.json:141]:
```suggestion
  "volta": {
```

**Review by secustor — APPROVED** [review]

**Pull request merged**